translate useContext referance (#423)
reactjs/tr.react.dev: src/content/reference/react/useContext.md

}
```

Bu değişiklik sonucunda, `MyApp` yeniden render edilse bile, `useContext(AuthContext)` çağıran bileşenler, `currentUser` değişmediği sürece yeniden render edilmeyecektir.

[`useMemo`](/reference/react/useMemo#skipping-re-rendering-of-components) ve [`useCallback`](/reference/react/useCallback#skipping-re-rendering-of-components) hakkında daha fazla bilgi edinin.

---

## Sorun Giderme {/*troubleshooting*/}

### Bileşenim sağlayıcının değerini görmüyor {/*my-component-doesnt-see-the-value-from-my-provider*/}

Bunun birkaç yaygın sebebi vardır:

1. `useContext()`'i çağırdığınız bileşenle aynı bileşende `<SomeContext.Provider>`bu bileşeni render ediyorsunuz. `SomeContext.Provider`'ı `useContext()` çağıran bileşenin üstüne ve dışına taşıyın.

> 2. Bileşeninizi `SomeContext.Provider` ile sarmalamayı unuttunuz veya düşündüğünüzden farklı bir yerde yerleştirdiniz. [React DevTools](/learn/react-developer-tools) kullanarak hiyerarşinin doğru olup olmadığını kontrol edin.

Reviewer: ```suggestion
2. Bileşeninizi `SomeContext.Provider` ile sarmalamayı unutmuş ya da ağacın istem dışı bir yerine yerleştirmiş olabilirsiniz. [React DevTools](/learn/react-developer-tools) kullanarak hiyerarşinin doğru olup olmadığını kontrol edin.
```

> 3. Sembolik bağlantılar kullanıyorsanız veya başka bir nedenle sağlayan bileşenin gördüğü `SomeContext`'ın, okuyan bileşenin gördüğü `SomeContext`'tan farklı olmasına neden olan bir yapılandırma sorunu ile karşılaşabilirsiniz. Bu, `window.SomeContext1` ve `window.SomeContext2` gibi global değişkenlere atayarak ve ardından konsolda `window.SomeContext1 === window.SomeContext2` gibi bir kontrol yaparak doğrulayabilirsiniz. Eğer aynı değillerse, yapılandırma sorununu düzeltin.

Reviewer: ```suggestion
3. Kullandığınız araçlardan kaynaklı, sağlayan bileşen tarafından görülen SomeContext ile okuyan bileşen tarafından görülen SomeContext nesnesinin iki farklı nesne olmasına neden olan derleme sorunlarıyla karşılaşabilirsiniz. Örneğin, sembolik bağlantılar kullanıyorsanız bu yaşanabilir. Bu durumu, `window.SomeContext1` ve `window.SomeContext2` gibi global değişkenlere atayıp konsolda `window.SomeContext1 === window.SomeContext2` kontrolü yaparak doğrulayabilirsiniz. Eğer aynı değillerse, derleme aracı seviyesinde sorunu düzeltin.
```

> ### Default değer farklı olsa da her zaman context'den `undefined` değerini alıyorum {/*i-am-always-getting-undefined-from-my-context-although-the-default-value-is-different*/}

Reviewer: ```suggestion
### Varsayılan değer farklı olsa bile context'den her zaman `undefined` alıyorum {/*i-am-always-getting-undefined-from-my-context-although-the-default-value-is-different*/}
```

> Ağacınızda `value` belirtmeksizin bir sağlayıcı olabilir:

Reviewer: ```suggestion
Ağaçta `value` verilmemiş bir sağlayıcı olabilir:
```

```js {4}
const MyContext = React.createContext('default');

function MyComponent() {
  const value = useContext(MyContext);
  // ...
}
```

> `value` belirtmeyi unutursanız, sanki `value={undefined}` geçirmişsiniz gibi davranılır.

Reviewer: ```suggestion
`value` belirtmeyi unutursanız, `value={undefined}` geçmiş gibi davranır .
```

> Ayrıca, farklı bir özellik adı yanlışlıkla kullanmış da olabilirsiniz:

Reviewer: ```suggestion
Yanlışlıkla farklı bir özellik adı kullanmış da olabilirsiniz:
```

```js {1,2}
// 🚩 Çalışmaz: özellik "value" olarak adlandırılmalı
<ThemeContext.Provider theme={theme}>
   <Button />
</ThemeContext.Provider>
```

> Bu durumlarda, React tarafından konsolda bir uyarı görmeniz gerektiğini unutmayın. Bunları düzeltmek için özelliği `value` olarak adlandırın:

Reviewer: ```suggestion
Her iki durumda da konsolda React uyarısı görmelisiniz. Bunları düzeltmek için prop'u `value` olarak adlandırın:
```

```js {1,2}

> // ✅ value özelliğini kullanarak geçirme

Reviewer: ```suggestion
// ✅ value prop geçirme
```

<ThemeContext.Provider value={theme}>
   <Button />
</ThemeContext.Provider>
```

> Önemli bir not olarak, [varsayılan değerinizin (`createContext` çağrısındaki `defaultValue`)](#specifying-a-fallback-default-value) yalnızca **hiçbir eşleşen sağlayıcı yoksa** kullanılacağına dikkat edin. Alt ağaçta bir `<SomeContext.Provider value={undefined}>` bileşeni varsa, `useContext(SomeContext)` çağrısını yapan bileşen `undefined` olarak context değerini alacaktır.

Reviewer: ```suggestion
Unutmayın, [`createContext(defaultValue)` çağrınızdaki varsayılan değer](#specifying-a-fallback-default-value) yalnızca **eşleşen hiçbir sağlayıcı yoksa** kullanılır. Alt ağaçta bir yerde `<SomeContext.Provider value={undefined}>` bileşeni varsa, `useContext(SomeContext)` çağrısını yapan bileşen context değeri olarak `undefined` *alacaktır*.
```
